stm32: i2c: i2c_api :  add  test config when interrupt mode is disabled

djiatsaf-st opened the pull request:
Following observation made in this PR https://github.com/zephyrproject-rtos/zephyr/pull/91184 

This PR introduces a test scenario in the `i2c_api` test driver for `build-only`, to check for CI failures when the interrupt mode is disabled (`CONFIG_I2C_STM32_INTERRUPT=n`).

depends on https://github.com/zephyrproject-rtos/zephyr/pull/91345

erwango commented on tests/drivers/i2c/i2c_api/testcase.yaml:
```diff
@@ -18,3 +18,26 @@ tests:
     extra_args:
       - DTC_OVERLAY_FILE="./boards/${BOARD}_sci_b_i2c.overlay"
       - CONF_FILE="./prj.conf ./boards/${BOARD}_sci_b_i2c.conf"
+  drivers.i2c.stm32_interrupt:
```
Name is not consistent with configuration.

djiatsaf-st replied:
name updated

erwango reviewed (changes requested):
Thanks but let's make sure we make it is efficient and easy to maintain

djiatsaf-st committed:
tests: drivers: i2c: i2c_api: add board overlays

add arduino_i2c node in overlay files to be enable building the i2c_api
test on multiple stm32 boards,compatible for both i2c V1 and V2
driver.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
tests: drivers: i2c: i2c_api: add a scenario for ci building

add scenario in build-only operation to check for CI failures
when interrupt is disabled for the STM32 i2c driver.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st commented on tests/drivers/i2c/i2c_api/boards/nucleo_c071rb.overlay:
```diff
@@ -0,0 +1,5 @@
+/ {
+	aliases {
+		i2c-0 =&i2c1;
```
Update with arduino_i2c and single file.

etienne-lms reviewed (commented):
LGTM.
In commit "tests: drivers: i2c: i2c_api: add a scenario for ci building",
replace  *"for the i2c driver."* with *"for the STM32 I2C driver."*.

erwango commented on tests/drivers/i2c/i2c_api/testcase.yaml:
```diff
@@ -18,3 +18,12 @@ tests:
     extra_args:
       - DTC_OVERLAY_FILE="./boards/${BOARD}_sci_b_i2c.overlay"
       - CONF_FILE="./prj.conf ./boards/${BOARD}_sci_b_i2c.conf"
+  drivers.i2c.stm32_interrupt.disabled:
```
Nitpick:
```
  drivers.i2c.stm32.interrupt_disabled:
```

erwango reviewed (changes requested):
Much better, thanks!

djiatsaf-st commented on tests/drivers/i2c/i2c_api/stm32_arduino.overlay:
done
